server/Controllers/GameControllers/snake.js

```diff
@@ -1,12 +1,115 @@
+const { SNAKE } = require('./gameConstants');
+
 class Snake {
-  constructor(playerId, initialPositionX, initialPositionY,
-    initialDirection, snakeColor, initialSnakeLength) {
+  /**
+   * @param  {string} playerId
+   * @param  {Object} initialProperties initial values for snake body and motion.
+   * @param  {Object} game instance of game class for status of current game. 
+   */
+  constructor(playerId, initialProperties, game) {
     this.ownerId = playerId;
-    this.head = { x: initialPositionX, y: initialPositionY };
-    this.direction = initialDirection;
-    this.color = snakeColor;
-    this.bodyLength = initialSnakeLength;
+    this.head = { x: initialProperties.position.x, y: initialProperties.position.y };
+    this.direction = initialProperties.direction;
+    this.velocity = initialProperties.velocity;
+    this.color = initialProperties.color;
+    this.length = initialProperties.length;
```

Comment:
If you goal is setting defaults, you can use spread or `Object.assign` to create a set of sensible defaults and create another object and overwrite `this` props.